### PR TITLE
Technical support function - Add share dialog (EXPOSUREAPP-6057)

### PR DESCRIPTION
**Pull request by LukasLechnerDev**
Instead of storing the log locally, the user should be able to share the error logs with a share sheet. 

image

### Testing: 
- App-Informations -> Error Reports -> Check if sharing the logs works
- Please also test if sharing of the QR-code poster still works, as some shared logic was touched

### CODE REVIEW

**Comment by AlexanderAlferov** [comment]
Tested on Samsung S8.
Logs sharing works great 👍 
Unfortunately, QR sharing feels off right now:
1) share button executes after second tap/press
2) after first sharing it remembers your choice and you could not change it (only reseting defaults for application you chose could help)

**Comment by AlexanderAlferov** [comment]
> Tested on Samsung S8.
> Logs sharing works great 👍
> Unfortunately, QR sharing feels off right now:
> 
> 1. share button executes after second tap/press
> 2. after first sharing it remembers your choice and you could not change it (only reseting defaults for application you chose could help)

It appears that this behaviour have nothing with this PR. We would need to test on more devices.

**Comment by mtwalli** [comment]
> Tested on Samsung S8.
> Logs sharing works great 👍
> Unfortunately, QR sharing feels off right now:
> 
> 1. share button executes after second tap/press
> 2. after first sharing it remembers your choice and you could not change it (only reseting defaults for application you chose could help)

this a video from S8: 

Uploading device-2021-04-19-163941.mp4…

**Comment by mtwalli** [comment]
> Tested on Samsung S8.
> Logs sharing works great 👍
> Unfortunately, QR sharing feels off right now:
> 
> 1. share button executes after second tap/press
> 2. after first sharing it remembers your choice and you could not change it (only reseting defaults for application you chose could help)

Video from S8, Is this what you expect @AlexanderAlferov ?
https://user-images.githubusercontent.com/25054729/115255439-8eb92d00-a12e-11eb-8942-0ae8844e4a3a.mp4

**Review by AlexanderAlferov — DISMISSED** [review]
Thanks for extra testing - It looks like I have local problems with sharing menu on my device.
LGTM 🚀

File: Corona-Warn-App/src/main/res/values-en/strings.xml [hunk 1]
```diff
@@ -852,10 +852,10 @@
     <string name="debugging_debuglog_action_start_recording">"Start"</string>
     <!-- XBUT: Button text to stop the log recording -->
     <string name="debugging_debuglog_action_stop_recording">"Stop and Delete"</string>
-    <!-- XBUT: Button text to share the log recording -->
+    <!-- XBUT: Button text to send the log recording to the server -->
     <string name="debugging_debuglog_action_share_log">"Send Error Report"</string>
-    <!-- XBUT: Button text to locally store the log recording -->
-    <string name="debugging_debuglog_action_local_log_store">"Save Locally and Continue"</string>
+    <!-- XBUT: Button text to export the log -->
+    <string name="debugging_debuglog_action_local_log_export">"Save Locally and Continue"</string>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/values-en/strings.xml:858]: should not we change that? it does not save anymore?

**Review by mtwalli — DISMISSED** [review]
LGTM, I think the text should change or ?

File: Corona-Warn-App/src/main/res/values-en/strings.xml (hunk 1 above)
Comment (harambasicluka) [reply Corona-Warn-App/src/main/res/values-en/strings.xml:858]: Valid point, I'll ping @janetback .

**Review by janetback — DISMISSED** [review]
UA reviewed

File: Corona-Warn-App/src/main/res/values-en/strings.xml (hunk 1 above)
Comment (harambasicluka) [reply Corona-Warn-App/src/main/res/values-en/strings.xml:858]: Discussed with @janetback changed export to share, will be part of the next translation delivery.

**Review by janetback — APPROVED** [review]
UA reviewed

File: Corona-Warn-App/src/main/res/values-en/strings.xml (hunk 1 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/res/values-en/strings.xml:858]: Yes, I only changed the german value according to the jira story description and expected the other languages in the next translation delivery.

**Review by AlexanderAlferov — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**